lib/djvAV/PFMRead.cpp

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2004-2020 Darby Johnston
+// All rights reserved.
+
+#include <djvAV/PFM.h>
+
+#include <djvSystem/File.h>
+#include <djvSystem/FileIO.h>
+#include <djvSystem/TextSystem.h>
+
+#include <djvCore/StringFormat.h>
+
+#include "djvSystem/FileIOFunc.h"
+
+using namespace djv::Core;
+
+namespace djv
+{
+    namespace AV
+    {
+        namespace IO
+        {
+            namespace PFM
+            {
+                Read::Read()
+                {}
+
+                Read::~Read()
+                {
+                    _finish();
+                }
+
+                std::shared_ptr<Read> Read::create(
+                    const System::File::Info& fileInfo,
+                    const ReadOptions& readOptions,
+                    const std::shared_ptr<System::TextSystem>& textSystem,
+                    const std::shared_ptr<System::ResourceSystem>& resourceSystem,
+                    const std::shared_ptr<System::LogSystem>& logSystem)
+                {
+                    auto out = std::shared_ptr<Read>(new Read);
+                    out->_init(fileInfo, readOptions, textSystem, resourceSystem, logSystem);
+                    return out;
+                }
+
+                Info Read::_readInfo(const std::string& fileName)
+                {
+                    auto io = System::File::IO::create();
+                    float scale;
+                    return _open(fileName, io, scale);
+                }
+
+                std::shared_ptr<Image::Data> Read::_readImage(const std::string& fileName)
+                {
+                    auto io = System::File::IO::create();
+                    float scale;
+                    const auto info = _open(fileName, io, scale);
+                    auto imageInfo = info.video[0];
+                    std::shared_ptr<Image::Data> out;
+                	
+#if defined(DJV_MMAP)
+                    out = Image::Data::create(imageInfo, io);
+#else // DJV_MMAP
+                    out = Image::Data::create(imageInfo);
+                    io->read(out->getData(), out->getDataByteCount());
+#endif // DJV_MMAP
+
+                	if(scale - 1 > 1E-6)
```

Comment:
It looks like there are some tabs here causing the formatting to appear out of place.